microdnf/main.cpp

```diff
@@ -343,6 +343,17 @@ static void set_commandline_args(Context & ctx) {
 
     add_new_named_arg_alias(*no_best, "nobest", "nobest", '\0', options_aliases_group, no_best_conflict_args);
 
+    {
+        auto exclude = ctx.get_argument_parser().add_new_named_arg("exclude");
+        exclude->set_long_name("exclude");
+        exclude->set_short_name('x');
+        exclude->set_short_description("exclude packages by name or glob");
+        exclude->set_has_value(true);
+        exclude->set_arg_value_help("package,...");
+        exclude->link_value(&config.excludepkgs());
```

Comment:
The "append" function is implemented by a user function in OptionBind.
One solution is to replace the direct `link_value` function with the user parse hook, which sets the option via option_binds.
This aproach is used in `setopt` and "--exclude" is a special case of setopt "setopt=excludepkgs=". But "setopt" is much more powerfull you can set excludes for specific repository.
```
exclude->set_parse_hook_func([&ctx](
                                 [[maybe_unused]] ArgumentParser::NamedArg * arg,
                                 [[maybe_unused]] const char * option,
                                 const char * value) {
    auto & conf = ctx.base.get_config();
    conf.opt_binds().at("excludepkgs").new_string(libdnf::Option::Priority::COMMANDLINE, value);
    return true;
});
```

Reply:
Thank you for your help